lib/graphiti/util/persistence.rb

```diff
@@ -86,13 +86,13 @@ def update_foreign_key(parent_object, attrs, x)
       if x[:sideload].polymorphic_has_one? || x[:sideload].polymorphic_has_many?
         attrs[:"#{x[:sideload].polymorphic_as}_type"] = nil
       end
-      attrs[x[:foreign_key]] = nil
+      @resource.adapter.set_foreign_key!(attrs, x[:foreign_key], nil)
```

Comment:
delegating foreign key management to adapter so that it is better handled in case of other type of databases.

Reply:
That's still a mystery to me. Is it guaranteed that the method won't be called on non AR-adapters? Because I don't see a stub. In any case this method name would not fit in abstract adapter.

Reply:
> Is it guaranteed that the method won't be called on non AR-adapters? 

In any relational DB related database it will be needed otherwise it's not needed 

>Because I don't see a stub. In any case this method name would not fit in abstract adapter.
hmm yes, I am changing this to work as `set_association` also I am trying to removing this whole method `update_foreign_key` it self and moving it to AR